### PR TITLE
feat: update mixin to use couchbaseX

### LINKED ISSUE #54: support for couchbaseX
# Issue
the loopback-cache mixin has a check to determine the version of couchbase connector being used, current the check only accepts couchbase3 and couchbase5. This results in it failing for couchbaseX ( which is actually couchbase5)

# Remedy 
- Update check to support couchbaseX
- update mixin to accept configurable TTL's

### PR DESCRIPTION
**Pull request by jakangah**
Fixes #54 

related to https://github.com/Wiredcraft/blue-internal/issues/479

### CODE REVIEW

File: test/utils.couchbase.test.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+require('should');
+const { isCouchbaseVersionSupported } = require('../utils/couchbase.js');
+
+describe('Couchbase Utils', () => {
+    describe('isCouchbaseVersionSupported', () => {
```
Comment (xavierchow) [inline test/utils.couchbase.test.js:7]: not sure why the indent goes with 4 spaces here, in the other places it's 2 spaces.

File: README.md [hunk 2]
```diff
@@ -63,6 +63,27 @@ To use the `couchbase` backend, setup a model using the [couchbase3 connector](h
 }
 ```
 
+To use a configurable loopback model as the ttl (only supported with couchbase currently) , omit the ttl option in the mixin configuration and use `lbEnvVar`, for example if we have a variable defined in our config as `cacheExpiryTTL` , we can use it as following within the mixin
```
Comment (xavierchow) [inline README.md:66]: I don't think we need this extra implementation, you can override the ttl from the Model method if needed, or maybe just document this case in README?
```
module.exports = Message => {
  disableRemoteMethods(Message);

  Message.mixin('CacheModel', {
    backend: 'couchbase',
    ttl: app.get('msgBucketTTL')
  });

  // omit the remaining ...
  
```

File: backends/couchbase.js [hunk 3]
```diff
@@ -1,30 +1,24 @@
...
-      // @see https://github.com/Wiredcraft/loopback-connector-couchbase3
-      if (/^couchbase[35]$/.test(ctx.Model.getConnector().name)) {
-        // @see http://docs.couchbase.com/sdk-api/couchbase-node-client-2.1.2/Bucket.html#touch
-        if (options.ttl > 30 * 24 * 60 * 60) {
-          ctx.options.expiry = convertToAbsExpiry(options.ttl);
-        } else {
-          ctx.options.expiry = options.ttl;
-        }
+      const connectorName = ctx.Model.getConnector().name;
+      if (isCouchbaseVersionSupported(connectorName)) {
```
Comment (xavierchow) [inline backends/couchbase.js:14]: 👍

File: test/utils.couchbase.test.js (hunk 1 above)
Comment (jakangah) [reply test/utils.couchbase.test.js:7]: @xavierchow will check the linting

File: README.md (hunk 2 above)
Comment (jakangah) [reply README.md:66]: @xavierchow hmm I considered this too, reason for not picking it was too abstract implementation to the mixin, I can remove the extra keyword if you prefer this

Comment (xavierchow) [reply README.md:66]: From the mixin's perspective, it cares about the TTL, it doesn't need to know if the TTL is a hardcoded number or an env var passed from the env var.
How to pass the TTL to the mixin is in the scope of the application and under the control of the application developer. I'd suggest removing the `lbEnvVar` usage.

Comment (jakangah) [reply README.md]: @xavierchow understood, but I think it's just another way to pass in a value my goal was to make it more flexible. I have removed it, please take another look, thanks

**Review by xavierchow — APPROVED** [review]

**Pull request merged**